test/hub/interactions/toggle-follow.spec.ts

```diff
@@ -0,0 +1,313 @@
+        await expect(
+          peripheryDataProvider.connect(userTwo).toggleFollow([FIRST_PROFILE_ID, FIRST_PROFILE_ID], [])
+        ).to.be.revertedWith(ERRORS.ARRAY_MISMATCH);
+      });
+
+      it('UserTwo should fail to toggle follow from a profile that has been burned', async function () {
+        await expect(lensHub.burn(FIRST_PROFILE_ID)).to.not.be.reverted;
+        await expect(
+          peripheryDataProvider.connect(userTwo).toggleFollow([FIRST_PROFILE_ID], [true])
+        ).to.be.revertedWith(ERRORS.TOKEN_DOES_NOT_EXIST);
+      });
+
+      it('UserTwo should fail to toggle follow for a followNFT that is not owned by them', async function () {
+        const followNFTAddress = await lensHub.getFollowNFT(FIRST_PROFILE_ID);
+        const followNFT = FollowNFT__factory.connect(followNFTAddress, user);
+
+        await expect(
+          followNFT.connect(userTwo).transferFrom(userTwoAddress, userAddress, 1)
+        ).to.not.be.reverted;
+
+        await expect(
+          peripheryDataProvider.connect(userTwo).toggleFollow([FIRST_PROFILE_ID], [true])
+        ).to.be.revertedWith(ERRORS.FOLLOW_INVALID);
+      });
+    });
+
+    context('Scenarios', function () {
+      it('UserTwo should toggle follow with true value, correct event should be emitted', async function () {
+        const tx = peripheryDataProvider.connect(userTwo).toggleFollow([FIRST_PROFILE_ID], [true]);
+
+        const receipt = await waitForTx(tx);
+
+        expect(receipt.logs.length).to.eq(1);
+        matchEvent(receipt, 'FollowsToggled', [
+          userTwoAddress,
+          [FIRST_PROFILE_ID],
+          [true],
+          await getTimestamp(),
+        ]);
+      });
+
+      it('User should create another profile, userTwo follows, then toggles both, one true, one false, correct event should be emitted', async function () {
+        await expect(
+          lensHub.createProfile({
+            to: userAddress,
+            handle: 'otherhandle',
+            imageURI: OTHER_MOCK_URI,
+            followModule: ZERO_ADDRESS,
+            followModuleData: [],
+            followNFTURI: MOCK_FOLLOW_NFT_URI,
+          })
+        ).to.not.be.reverted;
+        await expect(lensHub.connect(userTwo).follow([FIRST_PROFILE_ID + 1], [[]])).to.not.be.reverted;
+
+        const tx = peripheryDataProvider
+          .connect(userTwo)
+          .toggleFollow([FIRST_PROFILE_ID, FIRST_PROFILE_ID + 1], [true, false]);
+
+        const receipt = await waitForTx(tx);
+
+        expect(receipt.logs.length).to.eq(1);
+        matchEvent(receipt, 'FollowsToggled', [
+          userTwoAddress,
+          [FIRST_PROFILE_ID, FIRST_PROFILE_ID + 1],
+          [true, false],
+          await getTimestamp(),
+        ]);
+      });
+
+      it('UserTwo should toggle follow with false value, correct event should be emitted', async function () {
+        // const id = await getTokenId(userTwoAddress);
```

Comment:
Another comment that you probably want to remove here

Reply:
Done!